pkg/csr/csr.go

```diff
@@ -325,11 +325,7 @@ func findHostName(instanceInfo *instance.Info, instanceSigner ssh.Signer) (strin
 		return "", fmt.Errorf("error instantiating Windows instance: %w", err)
```

Comment:
Please add why this needs to be done to the commit message.
I believe you said it was so vsphere cloud controller can match the node to a VM.

Reply:
The description and comments of the linked bug expand on the why. I added more context to the description of the PR to avoid pushing a new commit and re-running all those that passed already. In case there is a code change request, I will also update the commit message.